code/modules/hydroponics/spreading/spreading.dm

```diff
@@ -0,0 +1,195 @@
+
+/obj/effect/plantsegment/examine(mob/user)
+	..()
+	if(!seed) return
+	var/traits = ""
+	if(seed.stinging) traits += "<span class='alert'>It's covered in tiny stingers.</span> "
+	if(seed.thorny) traits += "<span class='alert'>It's covered in sharp thorns.</span> "
+	if(seed.ligneous) traits += "It's a tough and hard vine that can't be easily cut. "
+	if(seed.carnivorous == 2) traits += "It's quivering viciously. "
+	if(seed.hematophage) traits += "It's roots are blood red... "
+	if(traits) user << traits
+
+/obj/effect/plantsegment/update_icon()
+	var/arbitrary_measurement_of_how_lush_I_am_right_now // a post-mortem dedication to Comic
+	if(harvest)
+		arbitrary_measurement_of_how_lush_I_am_right_now = 3
+	else if (age > mature_time)
+		arbitrary_measurement_of_how_lush_I_am_right_now = 2
+	else
+		arbitrary_measurement_of_how_lush_I_am_right_now = 1
+
+	var/at_fringe = get_dist(src,epicenter)
+	if(at_fringe >= round(spread_distance*0.9))
+		arbitrary_measurement_of_how_lush_I_am_right_now--
+	if(at_fringe >= round(spread_distance*0.7))
+		arbitrary_measurement_of_how_lush_I_am_right_now--
+	if(health < max_health)
+		arbitrary_measurement_of_how_lush_I_am_right_now -= round(-(health - max_health)/(max_health/3))
+
+	arbitrary_measurement_of_how_lush_I_am_right_now = max(1, arbitrary_measurement_of_how_lush_I_am_right_now)
+
+	switch(arbitrary_measurement_of_how_lush_I_am_right_now)
+		if(1)
+			icon_state = "Light[rand(1,3)]"
+			src.opacity = 0
+		if(2)
+			icon_state = "Med[rand(1,3)]"
+			src.opacity = 0
+		if(3)
+			icon_state = "Hvy[rand(1,3)]"
+			if(!limited_growth) src.opacity = 1
+
+	// Apply colour and light from seed datum.
+	if(seed.biolum && seed.biolum_colour)
+		set_light(1+round(seed.potency/20), l_color = seed.biolum_colour)
+		return
+	else
+		set_light(0)
+
+/obj/effect/plantsegment/attackby(var/obj/item/weapon/W, var/mob/user)
+
+	if(user.a_intent == I_HELP && (istype(W, /obj/item/weapon/wirecutters) || istype(W, /obj/item/weapon/scalpel)))
+		if(sampled)
+			user << "<span class='warning'>\The [src] has already been sampled recently.</span>"
+			return
+		if(!is_mature())
+			user << "<span class='warning'>\The [src] is not mature enough to yield a sample yet.</span>"
+			return
+		if(!seed)
+			user << "<span class='warning'>There is nothing to take a sample from.</span>"
+			return
+		if(sampled)
+			user << "<span class='danger'>You cannot take another sample from \the [src].</span>"
+			return
+		if(prob(70))
+			sampled = 1
+		seed.spawn_seed_packet(get_turf(user))
+		health -= (rand(3,5)*5)
+		sampled = 1
+		plant_controller.add_plant(src)
+	else
+		..()
+		take_damage(W)
+
+/obj/effect/plantsegment/proc/take_damage(var/obj/item/weapon/W)
+	if(!W.force) return 0
+	var/dmg = W.force
+	if(W.is_hot() || (W.is_sharp() && !seed.ligneous))
+		dmg = dmg*4
+	health -= dmg
+	check_health()
+	update_icon()
+
+/obj/effect/plantsegment/ex_act(severity)
+	switch(severity)
+		if(1.0)
+			die_off()
+			return
```

Comment:
Ditto 2: the dittoing.

<sup><sup>I wonder if it's dittoing or dittoening</sup></sup>